Original patch from gentoo: gentoo/src/patchsets/gcc/4.3.1/uclibc/90_all_304-index_macro.patch
diff -durN gcc-4.3.1.orig/libstdc++-v3/include/ext/rope gcc-4.3.1/libstdc++-v3/include/ext/rope
--- gcc-4.3.1.orig/libstdc++-v3/include/ext/rope	2008-01-31 19:44:55.000000000 +0100
+++ gcc-4.3.1/libstdc++-v3/include/ext/rope	2008-06-10 14:59:02.000000000 +0200
@@ -59,6 +59,9 @@
 #include <bits/gthr.h>
 #include <tr1/functional>
 
+/* cope w/ index defined as macro, SuSv3 proposal */
+#undef index
+
 # ifdef __GC
 #   define __GC_CONST const
 # else
diff -durN gcc-4.3.1.orig/libstdc++-v3/include/ext/ropeimpl.h gcc-4.3.1/libstdc++-v3/include/ext/ropeimpl.h
--- gcc-4.3.1.orig/libstdc++-v3/include/ext/ropeimpl.h	2007-05-04 17:06:46.000000000 +0200
+++ gcc-4.3.1/libstdc++-v3/include/ext/ropeimpl.h	2008-06-10 14:59:02.000000000 +0200
@@ -54,6 +54,9 @@
 #include <ext/memory> // For uninitialized_copy_n
 #include <ext/numeric> // For power
 
+/* cope w/ index defined as macro, SuSv3 proposal */
+#undef index
+
 _GLIBCXX_BEGIN_NAMESPACE(__gnu_cxx)
 
   using std::size_t;
